pca/utils/dependency_injection.py

```diff
@@ -9,26 +10,26 @@
 
 
 class Container:
-    def __init__(self):
+    def __init__(self, default_scope: 'Scopes'=None):
         self._registry = {}
-        self.default_scope = Scopes.INSTANCE
+        self._default_scope_function = default_scope
+
+    @staticmethod
+    def _get_registry_key(identifier: NameOrInterface, qualifier: Any=None) -> tuple:
+        return identifier, qualifier,
 
     def register_by_name(self, name: str, constructor: Callable):
         key = Container._get_registry_key(name)  # TODO: qualifier
         if key in self._registry:
             raise ValueError(f'Ambiguous name: {name}.')
         self._registry[key] = constructor
 
-    @staticmethod
-    def _get_registry_key(identifier: NameOrInterface, qualifier: Any=None) -> tuple:
-        return identifier, qualifier,
-
     def find_by_name(self, name: str) -> Any:
         key = Container._get_registry_key(name)  # TODO: qualifier
         return self.get_object(self._registry.get(key))
 
     def get_object(self, constructor: Callable) -> Any:
-        call = getattr(constructor, '__scope_type', self.default_scope)
+        call = getattr(constructor, '__scope_function', self._default_scope_function)
```

Comment:
I've meant: change `call` name to `scope_function`. `scope_type` and `default_scope` were ok.